tests/e2e/test_write.py

```diff
@@ -0,0 +1,103 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import random
+import string
+import pytest
+
+from dataclasses import dataclass
+from azstoragetorch.io import BlobIO
+from azstoragetorch.exceptions import FatalBlobIOWriteError
+
+_STAGE_BLOCK_SIZE = 32 * 1024 * 1024
+
+
+@dataclass
+class Blob:
+    data: bytes
+    url: str
+
+
+@pytest.fixture(scope="module")
+def small_blob(account_url, container_client):
+    return write_blob_url(account_url, container_client, sample_data(20))
+
+
+@pytest.fixture(scope="module")
+def large_blob(account_url, container_client):
+    return write_blob_url(
+        account_url, container_client, sample_data(_STAGE_BLOCK_SIZE * 2)
+    )
+
+
+@pytest.fixture
+def blob(request):
+    return request.getfixturevalue(f"{request.param}_blob")
+
+
+def random_resource_name(name_length=8):
+    return "".join(
+        random.choices(string.ascii_lowercase + string.digits, k=name_length)
+    )
+
+
+def write_blob_url(account_url, container_client, data):
+    blob_name = random_resource_name()
+    url = f"{account_url}/{container_client.container_name}/{blob_name}"
+
+    return Blob(data=data, url=url)
+
+
+def sample_data(data_length=20):
+    return os.urandom(data_length)
```

Comment:
It might make sense to try to consolidate the various versions that we have of `sample_data` and `random_resource_name` functions so that we can reuse it. I'm thinking we can introduce a `utils.py` file for end to end tests similar to what I did here for the unit tests: https://github.com/Azure/azure-storage-for-pytorch/blob/main/tests/unit/utils.py and we can then import directly from the `utils.py` for any shared end to end test utilities.